### PR TITLE
feat(internal): Massage registryUrls

### LINKED ISSUE #8651: Massage registryUrls
**What would you like Renovate to be able to do?**

Refactor registryUrls to add `https://` prefix if necessary.

**Did you already have any implementation ideas?**

The datasource index could check if the current string is a valid URL and if not then check if adding `https://` to it makes it valid.

### PR DESCRIPTION
**Pull request by renovate-testing**
## Changes:

Add additional check for urls without protocol, and assume `https` by default.

## Context:

Closes #8651

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
Can you add a check if the newly massaged URL is valid and if not then filter it out?

File: lib/datasource/crate/index.ts [hunk 1]
```diff
@@ -141,7 +141,7 @@ async function fetchRegistryInfo(
   let url: URL;
   try {
     url = new URL(registryUrl);
-  } catch (err) {
+  } catch (err) /* istanbul ignore next */ {
```
Comment (viceice) [inline lib/datasource/crate/index.ts:144]: this should have a test
```suggestion
  } catch (err) {
```

Comment (rarkins) [reply lib/datasource/crate/index.ts:144]: I guess it shouldn't happen and can be removed if we validate and URL in the index though?

Comment (viceice) [reply lib/datasource/crate/index.ts:144]: if it's validated before, we should remove the catch and let bubble any error through upper layers

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**